Explain our use of mypyc in the FAQ (#3002)
psf/black: docs/faq.md

errors is not a goal. It can format all code accepted by CPython (if you find an example
where that doesn't hold, please report a bug!), but it may also format some code that
CPython doesn't accept.

## What is `compiled: yes/no` all about in the version output?

While _Black_ is indeed a pure Python project, we use [mypyc] to compile _Black_ into a
C Python extension, usually doubling performance. These compiled wheels are available
for 64-bit versions of Windows, Linux (via the manylinux standard), and macOS across all
supported CPython versions.

> Platforms including musl-based and/or ARM linux distributions, and ARM Windows are

Reviewer: ```suggestion
Platforms including musl-based and/or ARM Linux distributions, and ARM Windows are
```

> currently **not** supported. These platforms will fallback to the (slower) Pure Python

Reviewer: ```suggestion
currently **not** supported. These platforms will fall back to the slower pure Python
```

wheel available on PyPI.

If you are experiencing exceptionally weird issues or even segfaults, you can try the

> Pure Python wheel by passing `--no-binary black` to your pip install invocation.

Reviewer: hm won't that use the sdist instead?

> Pure Python wheel by passing `--no-binary black` to your pip install invocation.

Author: ah shoot you're correct, I think it's still worth keeping this but I will note that this uses the sdist (since they're a bit more unreliable to install)

errors is not a goal. It can format all code accepted by CPython (if you find an example
where that doesn't hold, please report a bug!), but it may also format some code that
CPython doesn't accept.

## What is `compiled: yes/no` all about in the version output?

While _Black_ is indeed a pure Python project, we use [mypyc] to compile _Black_ into a
C Python extension, usually doubling performance. These compiled wheels are available
for 64-bit versions of Windows, Linux (via the manylinux standard), and macOS across all
supported CPython versions.

Platforms including musl-based and/or ARM Linux distributions, and ARM Windows are
currently **not** supported. These platforms will fall back to the slower pure Python
wheel available on PyPI.

If you are experiencing exceptionally weird issues or even segfaults, you can try

> passing `--no-binary black` to your pip install invocation. Note this flag excludes all
> wheels (including the pure Python wheel) so this will involve building the sdist, this
> may be a tad more unreliable.

Reviewer: ```suggestion
passing `--no-binary black` to your pip install invocation. This flag excludes all
wheels (including the pure Python wheel), so this command will use the sdist.
```
I don't think vague wording like "a tad more unreliable" is helpful. Maybe we can link to some packaging documentation about sdists?

Author: Resolved in https://github.com/psf/black/pull/3002/commits/2805958c2ee874f30c5b19474ef3df061c639c05
